UnityProject/Assets/Scripts/Player/PlayerScript.cs

```diff
@@ -575,6 +590,82 @@ public void OnMouseExit()
 		UIManager.SetToolTip = "";
 	}
 
+	public System.Random RNG = new System.Random();
+
+	public void OnInteract(TargetedInteraction Interaction, Component interactable)
+	{
+		if (Interaction != null)
+		{
+			if (ComponentManager.TryGetUniversalObjectPhysics(interactable.gameObject, out var UOP))
+			{
+				var Detail = UOP.attributes.Component.OrNull()?.AppliedDetail;
+				if (Detail == null) return;
+
+				if (RNG.Next(0, 100) > 55)
+				{
+
+					bool WearingGloves = false;
+					var slots = DynamicItemStorage.GetNamedItemSlots(NamedSlot.hands);
+					if (slots.Count != 0)
+					{
+						var slot = slots.PickRandom();
+
+						if (slot.Item != null)
+						{
+							WearingGloves = true;
+							Detail.AddDetail(new Detail()
+							{
+								CausedByInstanceID = slot.Item.gameObject.GetInstanceID(),
+								Description = $" A fibre from a {slot.Item.gameObject.ExpensiveName()}",
+								DetailType = DetailType.Fibre
+							});
+						}
+					}
+
+					if (WearingGloves == false) {
+						var slot = DynamicItemStorage.GetActiveHandSlot();
+						Detail.AddDetail(new Detail()
+						{
+							CausedByInstanceID = slot.ItemStorage.gameObject.GetInstanceID(),
+							Description = $" A fingerprint ",
+							DetailType = DetailType.Fingerprints
+						});
+					}
+				}
+
+
+
+				if (RNG.Next(0, 100) > 65)
+				{
+					var slots = DynamicItemStorage.GetNamedItemSlots(NamedSlot.uniform);
+					if (slots.Count != 0)
+					{
+						var slot = slots.PickRandom();
+						if (slot.Item != null)
+						{
+							Detail.AddDetail(new Detail()
+							{
+								CausedByInstanceID = slot.Item.gameObject.GetInstanceID(),
+								Description = $" A fibre from a {slot.Item.gameObject.ExpensiveName()}",
+								DetailType = DetailType.Fibre
+							});
+						}
+					}
```

Comment:
Duplicated code from above. I don't understand why we're doing the same thing again with another dice roll.

Reply:
I see, that makes more sense. It's still kinda duplicated, but not too bad if it is for something else.